### ISSUE TITLE
AgileMapper fails to populate member due to an unused constructor

### ISSUE THREAD

**Issue by bslbckr** [agileobjects/AgileMapper#139]
We observed an unexpected behavior, where AgileMapper failed to populate a target-member.
The problem is probably caused due to an equally named parameter of a constructor, that is not used at all during the mapping.

Our target class looks similar to this:
``` csharp
namespace Target {
	public class Entry {
		public Entry() {
			Name = "";
			Color = Color.Black;
		}
		
		public Entry(string name, Color color) {
			Name = name;
			Color = color;
		}
		
		public string Name {get;set;}
		public Color Color {get;set;}
		public string ColorString
		{
			get 
			{
				return Color != null ? TypeDescriptor.GetConverter(Color).ConvertToString(Color) : string.Empty;
			}
			set
			{
				Color = (Color) TypeDescriptor.GetConverter(typeof(Color)).ConvertFromString(value);
			}
		}
	}
}
```
We are mapping to `Target.Entry` from this source type:
``` csharp
namespace Source {
	public class Entry {
		public string Name {get;set;}
		public string ColorString {get;set;}
	}
}
```
The source type does not contain a matching entry for `Target.Entry.Color`, hence the only useable constructor is the parameter-less one. Nevertheless, AgileMapper writes in the mapping plan, that it cannot populate target-member `Name` as it is already populated by the constructor.

The complete example can be found in this fiddle: https://dotnetfiddle.net/Uu2y2c 

There is an easy workaround for this bug: Simply rename the constructor-parameter `name`. Nevertheless, the current behavior is confusing.

**Comment by SteveWilkes** [comment]
Well found! I'll sort it out. Thanks very much :)

**Comment by SteveWilkes** [comment]
This is fixed in the latest code - release to follow.

Cheers!

**Comment by SteveWilkes** [comment]
This is fixed as of v1.5-preview1, which I've just uploaded [to NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper); if the mapper can't be certain a constructor parameter with the same name as a member will be used, the member will be mapped.

Thanks again! :)
